src/rcc/rec.rs

```diff
@@ -246,6 +262,32 @@ macro_rules! peripheral_reset_and_enable_control {
                             }
                         )*
                     )*
+                    impl PeripheralREC {
+                        $(          // Group kernel clocks
+                            $(
+                                /// Modify the kernel clock for
+                                #[doc=$clk_doc_g]
+                                /// . See RM0433 Section 8.5.8.
+                                ///
+                                /// It is possible to switch this clock
+                                /// dynamically without generating spurs or
+                                /// timing violations. However, the user must
+                                /// ensure that both clocks are running. See
+                                /// RM0433 Section 8.5.10
+                                pub fn [< kernel_ $pk_g:lower _clk_mux >](&mut self, sel: [< $pk_g ClkSel >]) -> &mut Self {
```

Comment:
Would recommend referring to the `paste` crate to inform the reader, this is a heft macro.